signer/src/block_observer.rs

```diff
@@ -628,6 +634,38 @@ impl<C: Context, B> BlockObserver<C, B> {
         tracing::info!("updating the signer state with the current signer set");
         self.set_signer_set_and_aggregate_key(chain_tip).await
     }
+
+    /// Checks if the latest dkg share is pending and is no longer valid
+    async fn check_pending_dkg_shares(&self, chain_tip: BlockHash) -> Result<(), Error> {
+        let db = self.context.get_storage_mut();
+
+        let last_dkg = db.get_latest_encrypted_dkg_shares().await?;
+        let Some(
+            last_dkg @ EncryptedDkgShares {
+                dkg_shares_status: model::DkgSharesStatus::Unverified,
+                ..
+            },
+        ) = last_dkg
+        else {
+            return Ok(());
+        };
+
+        let chain_tip = db
+            .get_bitcoin_block(&chain_tip.into())
+            .await?
+            .ok_or(Error::NoChainTip)?;
+        let verification_window = self.context.config().signer.dkg_verification_window;
+
+        if last_dkg
+            .started_at_bitcoin_block_height
+            .saturating_add(verification_window as u64)
+            < chain_tip.block_height
```

Comment:
nit: for readability, I'd break this up into multiple lines like so:
```rust
let max_verification_height = last_dkg
    .started_at_bitcoin_block_height
    .saturating_add(verification_window as u64);

if max_verification_height < chain_tip.block_height {
```

Reply:
Done in https://github.com/stacks-network/sbtc/pull/1317/commits/bf615de39807eb21efad8ced41cb8d8ce27682c6